src/WatchServer/WatchServerClient.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\WatchServer;
+
+use Patchlevel\EventSourcing\Aggregate\AggregateChanged;
+
+use function base64_encode;
+use function fclose;
+use function restore_error_handler;
+use function serialize;
+use function set_error_handler;
+use function stream_socket_client;
+use function stream_socket_sendto;
+use function stream_socket_shutdown;
+use function strpos;
+
+use const STREAM_CLIENT_ASYNC_CONNECT;
+use const STREAM_CLIENT_CONNECT;
+use const STREAM_SHUT_RDWR;
+
+class WatchServerClient
+{
+    private string $host;
+
+    /** @var resource|null */
+    private $socket;
+
+    /**
+     * @param string $host The server host
+     */
+    public function __construct(string $host)
+    {
+        if (strpos($host, '://') === false) {
+            $host = 'tcp://' . $host;
+        }
+
+        $this->host = $host;
+        $this->socket = null;
+    }
+
+    public function send(AggregateChanged $event): bool
```

Comment:
Couldnt we get the singature to void instead of boolean and throw different exception for the different cases where the send was not successful instead only returning false?

Reply:
done